tracing/tracing/src/main/java/io/helidon/tracing/NoOpTracer.java

```diff
@@ -27,6 +28,28 @@ class NoOpTracer implements Tracer {
 
     private static final Scope SCOPE = new Scope();
 
+    private static final WritableBaggage EMPTY_BAGGAGE = new WritableBaggage() {
+        @Override
+        public Optional<String> get(String key) {
+            return Optional.empty();
+        }
+
+        @Override
+        public Set<String> keys() {
+            return Set.of();
+        }
+
+        @Override
+        public boolean containsKey(String key) {
+            return false;
+        }
+
+        @Override
+        public WritableBaggage set(String key, String value) {
```

Comment:
(Naïve question: must it be a `WriteableBaggage` implementation if it's just going to ignore this call anyway?)

Reply:
In this `NoOpTracer` the inner classes `Span` and `SpanContext` have a `baggage()` method declared by the interfaces they implement to return a `WriteableBaggage`. Each of the `baggage()` methods returns the`EMPTY_BAGGAGE` constant so it needs to implement `WriteableBaggage` even though, as you point out, it doesn't really write anything.